wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -195,13 +195,14 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
      */
 
     @Throws(Exception::class)
-    suspend fun searchUsernames(text: String, onlyExactUsername: Boolean = false): List<UsernameSearchResult> {
+    suspend fun searchUsernames(text: String, onlyExactUsername: Boolean = false, limit: Int = -1): List<UsernameSearchResult> {
         val userIdString = blockchainIdentity.uniqueIdString
         val userId = blockchainIdentity.uniqueIdentifier
 
         // Names.search does support retrieving 100 names at a time if retrieveAll = false
         //TODO: Maybe add pagination later? Is very unlikely that a user will scroll past 100 search results
-        val nameDocuments = platform.names.search(text, Names.DEFAULT_PARENT_DOMAIN, false)
+        val nameDocuments = platform.names.search(text, Names.DEFAULT_PARENT_DOMAIN,
+                false, limit)
```

Comment:
limit is set to 3 earlier in the process, but `limit` is supplied as the `startAt` argument of `search`

Currently, `search` does not support a `limit` parameter.

```
    /**
     * Searches for and returns a list of all name documents that match the given name based
     * on these criteria: starts with.  Contains is not supported
     * @param text String
     * @param parentDomain String
     * @param retrieveAll Boolean
     * @param startAtIndex Int
     * @return List<Documents>
     */
    fun search(text: String, parentDomain: String, retrieveAll: Boolean, startAtIndex: Int = 0): List<Document>
```

Reply:
if `startAtIndex` is set to 3, then no results are returned if there are less than 3 items that match the search. 

On the Search username screen, the `index` is set to 100, so the first 100 usernames are ignored.

Reply:
There are two solutions:
1. Prune search results to match the `limit`
2. add a `limit` parameter in the dashpay/platform library for `Names.search`

Reply:
Since dashpay PR https://github.com/dashevo/android-dashpay/pull/30/files#diff-4aaba3784c9fd97aaa67d814745186f6d8fdd8c6e7faa8118eb398103bb73211R235 adds a limit parameter, shouldn't your code have a `limit = limit`, otherwise limit is the value used for startAtIndex?

Though when I tested this PR, it did display three results, but not sure if they were the first three or the second three.

Reply:
Fixed on bb9ff717fb723cc2088b9acd8ab19245975db2c3